code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -2,104 +2,164 @@
 	name = "flamethrower"
 	desc = "You are a firestarter!"
 	icon = 'icons/obj/flamethrower.dmi'
-	icon_state = "flamethrowerbase"
+	icon_state = "flamethrower_base"
 	item_state = "flamethrower_0"
+
 	obj_flags = OBJ_FLAG_CONDUCTIBLE
-	force = 3.0
-	throwforce = 10.0
+	force = 3
+
+	throwforce = 10
 	throw_speed = 1
 	throw_range = 5
+
 	w_class = ITEM_SIZE_LARGE
 	origin_tech = "{'combat':1}"
 	material = /decl/material/solid/metal/steel
-	var/status = 0
+
+	var/fire_sound = 'sound/weapons/flamethrower.ogg'
+	/// Whether we have an igniter secured (screwdrivered) to us or not
+	var/secured = FALSE
 	var/throw_amount = 100
-	var/lit = 0	//on or off
-	var/operating = 0//cooldown
-	var/turf/previousturf = null
-	var/obj/item/weldingtool/weldtool = null
+	/// on or off
+	var/lit = FALSE
+	/// cooldown
+	var/operating = FALSE
+	var/turf/previous_turf = null
+	var/obj/item/weldingtool/welding_tool = null
 	var/obj/item/assembly/igniter/igniter = null
 	var/obj/item/tank/tank = null
 
+/obj/item/flamethrower/Initialize(mapload, welder)
+	. = ..()
+	if(welder)
```

Comment:
Please don't change arg names/ordering, preserve the original args and add a new one.
````
/obj/item/flamethrower/Initialize(var/ml, var/material_key, var/welder)
````
Then use `welder = fart` in the new() call. 
I know a bunch of stuff does mess with arg order and names, it's something we need to fix and avoid.